client/client.go

```diff
@@ -2766,6 +2766,8 @@ func (c *Client) allAllocs() map[string]*structs.Allocation {
 	allocs := make(map[string]*structs.Allocation, len(ars))
 	for _, ar := range ars {
 		a := ar.Alloc()
+		as := ar.AllocState()
+		a.ClientStatus = as.ClientStatus
```

Comment:
I don't think this is safe as you're modifying `ar.alloc`. I think you need to copy it first.

This also gives `allAllocs` pretty special behavior. It returns a view of each alloc not available anywhere else. At a minimum we should comment this unusual special case, but since there only appears to be a single caller I think we should probably scrap this method entirely and inline the behavior in the caller (or write a one-off function we can unit test).

Reply:
agreed with the criticism and the suggestion.

Reply:
@schmichael , updated as suggested.